src/main/kotlin/org/jitsi/nlj/util/RangedString.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright @ 2018 - Present, 8x8 Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.util
+
+/** Format a sequence of ints as a ranged string, e.g. "1, 3-8, 9-10" */
+/* TODO: it'd be nice to support this for any integer type but I don't know
+    if there's a good way to represent this.
+ */
+fun Iterator<Int>.joinToRangedString(
+    separator: CharSequence = ", ",
+    rangeSeparator: CharSequence = "-",
+    prefix: CharSequence = "",
+    postfix: CharSequence = "",
+    rangeLimit: Int = -1,
+    truncated: CharSequence = "..."
+): String {
+    val buffer = StringBuilder()
+
+    buffer.append(prefix)
+    var rangeCount = 0
+    var previous: Int? = null
+    var inRange = false
+    for (element in this) {
+        if (previous == null) {
+            buffer.append(element.toString())
+            rangeCount++
+            inRange = false
+        } else if (element == previous + 1) {
+            inRange = true
+        } else {
+            if (inRange) {
+                buffer.append(rangeSeparator).append(previous.toString())
+            }
+            inRange = false
+            buffer.append(separator)
+            rangeCount++
```

Comment:
I don't quite get what `rangeCount` is tracking.  It _doesn't_ appear to be tracking the number of elements in range, so at the least its name is confusing.

Reply:
It's counting the number of ranges we've printed.  If you hit `rangeLimit` (if that's set), it'll stop printing ranges and print `truncated` instead.  See the unit test for `rangeLimit`.